### ISSUE TITLE
Extract atomic weight from NIST/CIAAW

### ISSUE THREAD

**Issue by ppinard** [ppinard/pyxray#3]
Referring to issue #2, this issue is to populate the SQL database with atomic weight. Ideally, this should be done automatically from an available online resource.

The most official tabulation of atomic weight appears to come from the [CIAAW](http://www.ciaaw.org/). Unfortunately, their website does not directly provide an ASCII table of the values. NIST has an online database of atomic weight and isotopes which can be generated as an ASCII file ([website](http://www.nist.gov/pml/data/comp.cfm), [results](http://physics.nist.gov/cgi-bin/Compositions/stand_alone.pl?ele=Li&all=all&ascii=ascii2&isotype=some)). The information appears to come from the CIAAW/IUPAC.

The goal would therefore to write a parser to extract the atomic weight from the NIST output. I would proceed as follows:

1. Use the Python library [`requests`](http://docs.python-requests.org/en/master/) to download the NIST ASCII output
2. Parse each isotope, storing their `Atomic Number`, `Relative Atomic Mass` and `Isotopic Composition`
3. For each atomic number, multiply the relative atomic mass by the isotopic composition and sum over all isotopes.
4. Add values to the SQL database using the  `ElementAtomicWeightProperty` model (see `populate_element_symbol_table(engine)` in _pyxray/sql/default.py_ as an example)
5. Write some unit tests to compare the values with those appearing in the [NIST periodic table](http://www.nist.gov/pml/data/images/illo_for_2014_PT_1.PNG)

The code should go in a new function `populate_element_atomic_weight_table_(engine)` in _pyxray/sql/default.py_.

**Comment by ppinard** [comment]
I would modify the above procedure.

First, I think it is better to use the `Standard Atomic Weight` field to retrieve the atomic weight. Secondly, since the parser strategy has changed (see #4), a parser should be created for the NIST database in _pyxray/parser/nist.py_ and should return [ElementAtomicWeight](https://github.com/ppinard/pyxray/blob/sql/pyxray/property.py) objects.

**Comment by ppinard** [comment]
The commit f73c35ebdc2a46b9a736b6c982f0ec8f05cdd547 looks very good. I modified some small syntax issues in commit b2f214240b1dfb347a4b51fad5ea406c9fd8388c:

- I think we might have several parsers from NIST so I rename the bibtex key to _coursey2015_ instead of _nist_
- As per PEP8, uppercase words are for constants. I renamed `ATOMIC_WEIGHTS` to `atomic_weights`

Finally, I activated caching in commit cf011617305b0193dbb2e5479d533f5e21cac43e and added a unit test in 51f70bd687895ebb6794b2d7024637f9253b38d2.

Unless @fingenerf sees other issues, I would close this issue.
